dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java

```diff
@@ -0,0 +1,104 @@
+package com.dotmarketing.db;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Constants;
+import com.dotmarketing.util.Logger;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.net.URL;
+import javax.sql.DataSource;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+
+/**
+ * Singleton class that provides a datasource using a <b>db.properties</b> file configuration
+ * @author nollymar
+ */
+public class DBPropertiesDataSourceStrategy implements DotDataSourceStrategy {
+
+    private static final String DB_PROPERTIES_FILE_NAME = "db.properties";
+
+    private static File propertiesFile;
+
+    @VisibleForTesting
+    DBPropertiesDataSourceStrategy(final File file){
+        propertiesFile = file;
+    }
+
+    private DBPropertiesDataSourceStrategy(){
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        final URL resourceURL = loader.getResource(DB_PROPERTIES_FILE_NAME);
+        if (resourceURL!=null){
+            propertiesFile = new File(resourceURL.getPath());
+        }
+
+    }
+
+    private static class SingletonHelper{
+        private static final DBPropertiesDataSourceStrategy INSTANCE = new DBPropertiesDataSourceStrategy();
+    }
+
+    public static DBPropertiesDataSourceStrategy getInstance(){
+        return SingletonHelper.INSTANCE;
+    }
+
+    /**
+     *
+     * @return True if a <b>db.properties</b> file exists in WEB-INF/classes directory
+     */
+    public boolean existsDBPropertiesFile() {
+        return propertiesFile!=null && propertiesFile.exists();
+    }
+
+    @Override
+    public DataSource apply() {
+        final PropertiesConfiguration properties = new PropertiesConfiguration();
+        try {
+
+            if (!(existsDBPropertiesFile())){
+                throw new FileNotFoundException("DB properties file not found");
+            }
+
+            properties.load(new FileInputStream(propertiesFile));
+
+            final HikariConfig config = getHikariConfig(properties);
+
+            properties.clear();
+            return new HikariDataSource(config);
+        } catch (ConfigurationException | FileNotFoundException e) {
+            Logger.error(DBPropertiesDataSourceStrategy.class,
+                    "---------- Error getting dbconnection " + Constants.DATABASE_DEFAULT_DATASOURCE
+                            + " from db.properties file",
+                    e);
+
+            throw new DotRuntimeException(e.toString());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [New exception is thrown in catch block, original stack trace may be lost](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5026767)